src/bucket.js

```diff
@@ -1971,6 +2006,22 @@ Bucket.prototype.setStorageClass = function(storageClass, options, callback) {
 };
 
 /**
+ * Set a user project to be billed for all requests made from this Bucket
+ * object and any files referenced from this Bucket object.
+ *
+ * @param {string} userProject The user project.
+ *
+ * @example
+ * var storage = require('@google-cloud/storage')();
+ * var bucket = storage.bucket('albums');
+ *
+ * bucket.setUserProject('grape-spaceship-123');
+ */
+Bucket.prototype.setUserProject = function(userProject) {
+  this.userProject = userProject;
```

Comment:
@frankyn I might be crazy, but I thought I remember reading in an early Requester Pays design requirement doc that we shouldn't store a `userProject` on a Bucket instance. Instead, it has to be supplied each time a method is called. Am I making that up?

Reply:
To elaborate, this would allow this kind of thing:

```js
var bucket = gcs.bucket('some-bucket', { userProject: 'my-project' })
bucket.getMetadata() // `userProject` passed to API automatically
```

As opposed to:

```js
var bucket = gcs.bucket('some-bucket')
bucket.getMetadata({ userProject: 'my-project' })
```

Reply:
Oh, that might be true, I never saw any of the design docs for this.

Reply:
@stephenplusplus I recall that a user needs to opt-in by specifying a user project. I don't recall seeing a design document.
However,  Ruby and PHP for example allow the user to opt into UserProject once when getting the bucket resource and follow-up operations using this bucket resource continue to use this value.

Reply:
@stephenplusplus based on that, do you think we should consider removing the query string parameter and only support the bucket option?

Reply:
@callmehiphop What did you mean by removing the query string parameter?

Reply:
Sorry, what I meant was that we should remove the option to specify a user project at the method level and only allow users to configure a user project at the bucket level.

Reply:
I think having both options is helpful when a service account has access to multiple project ids used for billing. Another reason is `UserProject` can be set for all Storage API requests and not only for a Requester Pays bucket.